### PR TITLE
Pr/content flatcard props update

### LINKED ISSUE #356: Update FlatCard required props and Content and FlatCard props for image and link
# Feature request

Currently, images are required for FlatCard component and validated with PropTypes but it's not always needed.

Content components sometimes use background images and there's no way to easily set background images. FlatCard components constantly use links but there's no way to easily append a link button.

**Describe the solution you'd like**

Remove the FlatCard requirement/validation and the DOM output if no image property is passed. Also, update the unit tests for FlatCard to reflect the changes. And add a prop for passing links to append a link button to FlatCard component.

For Content components: add a prop for passing an image source to set a background image.

### PR DESCRIPTION
**Pull request by juliantrueflynn**
# Description of changes

- Add optional prop `backgroundImageSource` to Content component
- Add optional prop `button` to FlatCard component
- Remove required prop `imageAlt` and `imageSource` from FlatCard component

# Issue Resolved

Fixes #356

### CODE REVIEW

**Comment by kylemh** [comment]
🤖 **This is a bot** 🤖<br /><br />🎉 Deployed Storybook preview! 🎉<br /><br />Click the link at the bottom of this comment to see it live.

Built with commit 423c66f84ce810699b72700799d36b0b6df2a0df

https://deploy-preview-357--operation-code-storybook.netlify.com

File: components/Content/__stories__/Content.stories.js [hunk 1]
```diff
@@ -40,6 +43,7 @@ storiesOf('Content', module)
     'with one column',
     withInfo()(() => (
       <Content
+        backgroundImageSource={text('imageSource', backgroundImage)}
```
Comment (kylemh) [inline components/Content/__stories__/Content.stories.js]: I like leaving the story to match up with default props. Also the knob name should match the prop name.

If you would please remove line 11 and then make the other three diffs on this page match this:
```
backgroundImageSource={text('backgroundImageSource', '')}
```

File: components/Cards/FlatCard/FlatCard.js [hunk 2]
```diff
@@ -4,28 +4,35 @@ import classNames from 'classnames';
...
-          <img className={styles.image} src={imageSource} alt={imageAlt} />
-        </div>
+        {imageSource && (
+          <div className={styles.rowCenter}>
+            <img className={styles.image} src={imageSource} alt={imageAlt} />
+          </div>
+        )}
         {header && <hr className={styles.divider} />}
         <div className={styles.children}>{children}</div>
+        {Button}
```
Comment (kylemh) [inline components/Cards/FlatCard/FlatCard.js]: I thought you would put the button into a div and then provide an opinionated positioning of any/all buttons that come into place?

Comment (juliantrueflynn) [reply components/Cards/FlatCard/FlatCard.js]: Sorry, just re-read convo from 2/26 in Slack and saw you mentioned opinionated positioning. To make sure we're on same page: parent div to button and a prop with choices `left`, `center`, `right` for positioning?

File: components/Content/__stories__/Content.stories.js (hunk 1 above)
Comment (juliantrueflynn) [reply components/Content/__stories__/Content.stories.js]: Okay, updating to that. FYI though, HeroBanner.stories.js doesn't have correctly named knob then, it also uses `imageSource` when prop is named `backgroundImageSource`.

File: components/Cards/FlatCard/FlatCard.js (hunk 2 above)
Comment (kylemh) [reply components/Cards/FlatCard/FlatCard.js]: no choices for positioning; always center.

File: components/Cards/FlatCard/__stories__/__snapshots__/FlatCard.stories.storyshot [hunk 3]
```diff
@@ -30,6 +30,7 @@ exports[`Storyshots Cards/FlatCard default 1`] = `
             veterans and spouses, and as a former captain in the Army was recipient of the Lt. Rowan
             Award, Meritorious Service Medal and Army Commendation Medal.          
     </div>
+    )
```
Comment (kylemh) [inline components/Cards/FlatCard/__stories__/__snapshots__/FlatCard.stories.storyshot]: looks like you added a closing parenthesis

File: components/Content/__stories__/Content.stories.js (hunk 1 above)
Comment (kylemh) [reply components/Content/__stories__/Content.stories.js]: Damn. Would you mind fixing that here?

File: components/Cards/FlatCard/__tests__/FlatCard.test.js [hunk 4]
```diff
@@ -18,14 +17,25 @@ describe('FlatCard', () => {
...
+    imageAlt: "David Molina's face",
+    ...requiredProps,
+  };
+
   it('should render with required props', () => {
     createSnapshotTest(<FlatCard {...requiredProps} />);
   });
 
+  it('should render with image props', () => {
+    createSnapshotTest(<FlatCard {...imageProps} />);
```
Comment (kylemh) [inline components/Cards/FlatCard/__tests__/FlatCard.test.js]: We still need to pass required props here

Comment (juliantrueflynn) [reply components/Cards/FlatCard/__tests__/FlatCard.test.js]: `imageProps` object has `requiredProps` properties by using spread operator right now (line 23). If that's hard to read I can refactor to `<FlatCard {...imageProps} {...requiredProps} />` on line 31.

Comment (kylemh) [reply components/Cards/FlatCard/__tests__/FlatCard.test.js]: Missed that! Thanks

**Pull request merged**